### PR TITLE
7340 remote config fix

### PR DESCRIPTION
**Pull request by prkeshri**
With reference to https://github.com/backstage/backstage/pull/8236
Fixes post review by Rugvip
1. Ensuring that reloadIntervals, if present, is a valid positive number
2. Remote config error message fix
3. Removed remote config from example backend
4. Lessened the writing.md
5. Multiple changesets

## Hey, I just made a Pull Request!

Backend was giving error whenever any remote url config was passed while starting it.

Ex: `yarn start --config ../../app-config.yaml --config https://some.domain.io/app-config.yaml`

Error: 
```
Backend failed to start up Error: Remote config detected but this feature is turned off
    at loadConfig (webpack-internal:///../config-loader/src/loader.ts:156:11)
    at loadBackendConfig (webpack-internal:///../backend-common/src/config.ts:214:100)
    at async main (webpack-internal:///./src/index.ts:100:18)
```
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages.
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes: Not present as there was no test written earlier as well. 
- [ ] Screenshots attached (for UI changes): No UI Change
- [x] All your commits have a `Signed-off-by` line in the message.

### CODE REVIEW

File: .changeset/brave-impalas-switch.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+These changes are needed in `packages/backend/src/index.ts` if remote urls are desired to be passed in --config option and read and watch remote files for config.
+
+```diff
+@@ -86,7 +86,11 @@ async function main() {
+   const config = await loadBackendConfig({
+     argv: process.argv,
+     logger,
++    remote: {
++      reloadIntervalSeconds: 60 * 60 * 12 // Check remote config changes every 12 hours. Change to your desired interval in seconds
```
Comment (Rugvip) [inline .changeset/brave-impalas-switch.md:15]: Perhaps we should have an example here that's a lil bit lower? :p maybe refresh every 10 min or so

File: packages/config-loader/src/loader.ts [hunk 2]
```diff
@@ -126,8 +126,16 @@ export async function loadConfig(
     .filter((e): e is { url: string } => e.hasOwnProperty('url'))
     .map(configTarget => configTarget.url);
 
-  if (remote === undefined && configUrls.length > 0) {
-    throw new Error(`Remote config detected but this feature is turned off`);
+  if (remote === undefined) {
+    if (configUrls.length > 0) {
+      throw new Error(
+        `Please make sure you are passing the remote option when loading the configuration. See https://backstage.io/docs/conf/writing#configuration-files for detailed info.`,
```
Comment (Rugvip) [inline packages/config-loader/src/loader.ts:132]: "...when loading remote configurations..." or something like that perhaps? just wanna avoid confusion

**Review by Rugvip — COMMENTED** [review]
Thanks, looks good! 👍 

Couple of CI checks that need resolving. The Markdown spell check can often be fixed by wrapping things up in `` `inline` `` code blocks, or it may be necessary to add an exception to `.github/styles/vocab.txt`

File: packages/config-loader/src/loader.ts (hunk 2 above)
Comment (prkeshri) [reply packages/config-loader/src/loader.ts:132]: Fixed.

File: .changeset/brave-impalas-switch.md (hunk 1 above)
Comment (prkeshri) [reply .changeset/brave-impalas-switch.md:15]: Done. 10 minutes seem too often but anyway it's an example and actual implementation is totally user dependent :D.

**Comment by prkeshri** [comment]
@Rugvip Have updated. Kindly check. Thanks.

**Comment by prkeshri** [comment]
Hi @benjdlambert / @Rugvip 

I can see that the check [[Link](https://github.com/backstage/backstage/runs/4508319061?check_suite_focus=true)] is displaying errors for several undocumented codes. Please let me know if anything is required from my end regarding these.

Thanks
Praveen

**Review by freben — APPROVED** [review]
yep the top and bottom of that build output says

```
*************************************************************************************
* You have uncommitted changes to the public API of a package.                      *
* To solve this, run `yarn build:api-reports` and commit all api-report.md changes. *
*************************************************************************************
```

i took the liberty of adding the updated report, things should go green now i think, looking good!

**Pull request merged**